frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -123,6 +148,19 @@ private boolean isBranchNameRepeated(String fileContent, String branchName) {
     return lines.stream().filter(line -> line.equals(branchName)).count() > 1;
   }
 
+  private String getParentBranchName(Path macheteFilePath, String branchName) throws BranchLayoutException {
```

Comment:
While you're at this, pls also get rid of custom parser from the recent code for detecting duplicate branches (can be on a separate PR)

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1272